schema/schema_test.go

```diff
@@ -0,0 +1,389 @@
+package schema
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path"
+	"reflect"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestCollectionSchema(t *testing.T) {
+	// invalid
+	type empty struct {
+	}
+
+	// invalid
+	type noPK struct {
+		Key1 string `json:"key_1"`
+	}
+
+	// valid. implicit one part primary key with index 1
+	type pk struct {
+		Key1 string `json:"key_1" tigrisdb:"primary_key"`
```

Comment:
what is the behavior if there is no JSON tag i.e. `json:"key_1"`

Reply:
Field name, "Key1" in this case.